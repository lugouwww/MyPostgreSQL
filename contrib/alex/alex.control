# alex extension
comment = 'alex access method - signature file based index'
default_version = '1.0'
module_pathname = '$libdir/alex'
relocatable = true
