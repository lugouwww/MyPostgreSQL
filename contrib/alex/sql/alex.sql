CREATE EXTENSION alex;

CREATE TABLE tst (
	i	int4,
    t	text
);

-- INSERT INTO tst SELECT i%1000000, '1' FROM generate_series(1,10000000) i;
INSERT INTO tst SELECT FLOOR(random() * 1000000)::int, '1' FROM generate_series(1,10000) i;
CREATE INDEX alexidx ON tst USING alex (i);

SET enable_seqscan=on;
SET enable_bitmapscan=off;
SET enable_indexscan=off;

EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 7;
SELECT count(*) FROM tst WHERE i = 7;
SELECT * FROM tst WHERE i = 7;

SET enable_seqscan=off;
SET enable_bitmapscan=on;
SET enable_indexscan=on;

EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 7;
SELECT count(*) FROM tst WHERE i = 7;
SELECT * FROM tst WHERE i = 7;
