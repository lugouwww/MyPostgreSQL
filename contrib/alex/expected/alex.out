CREATE EXTENSION alex;
CREATE TABLE tst (
	i	int4
);
INSERT INTO tst SELECT i%10 FROM generate_series(1,2000) i;
CREATE INDEX alexidx ON tst USING alex (i);

SET enable_seqscan=on;
SET enable_bitmapscan=off;
SET enable_indexscan=off;
SELECT count(*) FROM tst WHERE i = 7;
 count 
-------
   200
(1 row)

SET enable_seqscan=off;
SET enable_bitmapscan=on;
SET enable_indexscan=on;
EXPLAIN (COSTS OFF) SELECT count(*) FROM tst WHERE i = 7;
                QUERY PLAN                 
-------------------------------------------
 Aggregate
   ->  Bitmap Heap Scan on tst
         Recheck Cond: (i = 7)
         ->  Bitmap Index Scan on alexidx
               Index Cond: (i = 7)
(5 rows)

SELECT count(*) FROM tst WHERE i = 7;
 count 
-------
   200
(1 row)
